pkg/services/ngalert/state/historian/encode.go

```diff
@@ -0,0 +1,106 @@
+package historian
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	"github.com/grafana/grafana/pkg/components/loki/logproto"
+	"github.com/prometheus/common/model"
+	"golang.org/x/exp/slices"
+)
+
+type JsonEncoder struct{}
```

Comment:
nit: these structs are public but their methods are private, not sure if they should be.

Reply:
I made the types public so the anything wishing to use this as a library can pass in the right encoding. Such systems should not need to call any encoder methods directly, only pass one in